kubectl-volsync/cmd/migration_create_test.go

```diff
@@ -0,0 +1,281 @@
+package cmd
+
+import (
+	"context"
+	"fmt"
+	"time"
+
+	volsyncv1alpha1 "github.com/backube/volsync/api/v1alpha1"
+	. "github.com/onsi/ginkgo"
+	. "github.com/onsi/gomega"
+	"github.com/spf13/cobra"
+	corev1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/types"
+	"k8s.io/apimachinery/pkg/util/wait"
+	"k8s.io/klog/v2"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+)
+
+var _ = Describe("migration", func() {
+	cmd := &cobra.Command{}
+	init := false
+	var migrationCmdArgs = make(map[string]string)
+	mc := &migrationCreate{}
+	mrd := &migrationRelationshipDestination{}
+	ns := &corev1.Namespace{}
+	var k8sClient client.Client
+
+	BeforeEach(func() {
+		migrationCmdArgs = map[string]string{
+			"capacity":    "2Gi",
+			"pvcname":     "dest/volsync",
+			"accessmodes": "ReadWriteOnce",
+			"servicetype": "ClusterIP",
+		}
+
+		if init == false {
+			initMigrationCreateCmd(cmd)
+			cmd.Flags().String("relationship", "test", "")
+			cmd.Flags().String("config-dir", "/root/.volsync", "")
+
+			cobra.CheckErr(cmd.Execute())
+			mr, err := newMigrationRelationship(cmd)
+			Expect(err).ToNot(HaveOccurred())
+			Expect(mr).ToNot(BeNil())
+
+			mc.mr = mr
+			init = true
+		}
+
+		err := migrationCmdArgsSet(cmd, migrationCmdArgs)
+		Expect(err).ToNot(HaveOccurred())
+
+		err = mc.parseCLI(cmd)
+		Expect(err).NotTo(HaveOccurred())
+
+		k8sClient, err = newClient("")
+		Expect(err).ToNot(HaveOccurred())
+		Expect(k8sClient).ToNot(BeNil())
+		mc.clientObject = k8sClient
+
+	})
+
+	AfterEach(func() {
+		if ns.Name != "" {
+			Expect(mc.deleteNamespace(cmd.Context(), ns)).To(Succeed())
+		}
+	})
+
+	It("Verify migration create arguments: fails with missing pvcname arg, with no pre-existing PVC", func() {
+		err := cmd.Flags().Set("pvcname", "")
+		Expect(err).NotTo(HaveOccurred())
+		err = mc.parseCLI(cmd)
+		Expect(err).To(HaveOccurred())
+	})
+
+	It("Verify migration create arguments: works with minimum set of arguments, with no preexsting PVC ", func() {
+		err := mc.parseCLI(cmd)
+		Expect(err).NotTo(HaveOccurred())
+	})
+
+	It("Verify migration create arguments: fails with wrong accessMode arg value", func() {
+		err := cmd.Flags().Set("accessmodes", "Read-Write-Once")
+		Expect(err).NotTo(HaveOccurred())
+		err = mc.parseCLI(cmd)
+		Expect(err).To(HaveOccurred())
+	})
+
+	It("Verify migration create arguments: fails with wrong capacity arg value", func() {
+		err := cmd.Flags().Set("capacity", "2GB")
+		Expect(err).NotTo(HaveOccurred())
+		err = mc.parseCLI(cmd)
+		Expect(err).To(HaveOccurred())
+	})
+
+	It("Verify namespace creation with no pre-existing namespace", func() {
+		mrd.Namespace = mc.Namespace
+		ns, err := mc.ensureNamespace(cmd.Context())
+		Expect(err).ToNot(HaveOccurred())
+		Expect(ns).NotTo(BeNil())
+
+	})
+
+	It("Verify PVC creation with/without pre-existing PVC", func() {
```

Comment:
I'm having trouble figuring out what portion of the code is being tested here. There seems to be a good bit of duplication from `Run()`. Experiment: Can the code or test be restructured to make it more clear?

Reply:
Okay. I think I need to test functions than functionality. Am I right?

Reply:
Correct.
For full functional testing, we will add cli-based tests to the e2e suite.

Reply:
sure